geocity/apps/accounts/users.py

```diff
@@ -52,67 +52,50 @@ def get_integrator_permissions():
+    if remove_anonymous:
+        for qs_user in qs:
+            if qs_user.userprofile.is_anonymous:
+                anonymous_users.append(qs_user.pk)
+        qs = qs.exclude(pk__in=anonymous_users)
 
-    email_domains = [
-        domain.strip()
-        for domain in user_integrator_group.permit_department.integrator_email_domains.split(
-            ","
-        )
-    ]
-    emails = [
-        email.strip()
-        for email in user_integrator_group.permit_department.integrator_emails_exceptions.split(
-            ","
+    # Integrators can only view users for restricted email domains.
+    if not user.is_superuser:
+        user_integrator_group = user.groups.get(
+            permit_department__is_integrator_admin=True
         )
-    ]
 
-    qs = (
-        User.objects.annotate(
-            email_domain=Substr("email", StrIndex("email", Value("@")) + 1),
+        email_domains = [
+            domain.strip()
+            for domain in user_integrator_group.permit_department.integrator_email_domains.split(
+                ","
+            )
+        ]
+
+        emails = [
+            email.strip()
+            for email in user_integrator_group.permit_department.integrator_emails_exceptions.split(
+                ","
+            )
+        ]
+
+        qs = (
+            qs.annotate(
+                email_domain=Substr("email", StrIndex("email", Value("@")) + 1),
+            )
+            # hide anynomous user not belonging to the actual integrator
+            .filter(
+                Q(is_superuser=False),
+                Q(email_domain__in=email_domains) | Q(email__in=emails),
+                Q(groups__permit_department__integrator=user_integrator_group.pk)
+                | Q(groups__isnull=True)
+                | Q(groups__permit_department__is_integrator_admin=True),
+            )
+            .exclude()
+            .distinct()
         )
-        # hide anynomous user not belonging to the actual integrator
-        .filter(
-            Q(is_superuser=False),
-            Q(email_domain__in=email_domains) | Q(email__in=emails),
-            Q(groups__permit_department__integrator=user_integrator_group.pk)
-            | Q(groups__isnull=True)
-            | Q(groups__permit_department__is_integrator_admin=True),
-        )
-        .exclude()
-        .distinct()
-    )
-    integrator_administrative_entities_list = (
-        models.AdministrativeEntity.objects.associated_to_user(user).values_list(
-            "pk", flat=True
-        )
-    )
-
-    # Used to remove anonymous users from the list
-    anonymous_users = []
-    for user in qs:
-        if remove_anonymous and user.userprofile.is_anonymous:
-            anonymous_users.append(user.pk)
-        elif (
-            user.userprofile.is_anonymous
-            and user.userprofile.administrative_entity.pk
-            not in integrator_administrative_entities_list
-        ):
-            anonymous_users.append(user.pk)
```

Comment:
Only part deleted, everything works the same. With DRY approach. 

Reason to remove this : We want to remove anonymous_users anyway, there's no reason to show anonymous users.

Reply:
Actually it is easier to read putting the 2 codes side by side, as I changed the order of almost everything